docs.md

```diff
@@ -1,23 +1,476 @@
+      "properties": {
+         "input_property": "request_path",
+         "value": ["v1", "collections", "*"],
+         "exceptional_value": "obs",
+      },
+   }
+   ```
+   In the json above, the `value` key holds an asterik in the values section, to indicate that the endpoint x is allowed to do y. The `exceptional_value` key holds the value of the path parameter that is to be exempted.
+
+   The REGO equivalent of the above rule object is: <br />
+   `input.request_path[0] == "v1" && input.request_path[1] == "collections" && input.request_path[2] != "obs"`
+
+   ### -  Handling equality check between a particular property on the request object and a value: <br />
+   This logic handles cases where a particular property on the input object is to be checked for equality against a value. 
+   Example:
+
+   ```json
+   {
+      "command": "input_prop_equals",
+      "properties": {
+         "input_property": "company",
+         "value": "Geobeyond srl"
+      }
+   }
+   ```
+   In the json above, the `value` key holds the value that is to be checked for equality, while the `input_property` key holds the property that is to be checked in the input object.
+
+   The REGO equivalent of the above rule object is: <br />
+   `input.company == "Geobeyond srl"`
+
+
+
+ ### -  Allow access to a particular path: <br />
+ This logic handles cases where a specific path is to granted access, if certain property is present on the input object.
+
+   Example:
+   ```json
+   [
+      {
+         "command": "input_prop_equals",
+         "properties": {
+            "input_property": "request_path",
+            "value": ["v1", "collections", "lakes"],
+         },
+      },
+       {
+         "command": "input_prop_equals",
+         "properties": {
+            "input_property": "request_path",
+            "value": "admin",
+         },
+      }
+   ]
+   ```
+In the json above, the list of objects indicate a very speciall `allow` block, that combines two commands. The `input_property` key holds the property that is to be va;idated before the request to that path is passed. 
+
+The REGO equivalent of the above rule object is: <br />
+   `allow {
+  input.request_path == ["v1", "collections", "lakes", ""] && input.groupname == "admin"
+}` 
+
+   ## 2. Input_props_in
+   This logic checks if a particular property on the input object is present in a list of values from the database. <br />
+
+   Example:
+
+   ```json
+   {
+      "command": "input_prop_in",
+      "properties": {
+         "input_property": "company",
+         "datasource_name": "items",
+         "datasource_loop_variable": "name"
+      }
+}
+   ```
+   In the json above, the `input_property` key holds the property that is to be validated before the request to that path is passed. The `datasource_name` key holds the name of the datasource(a list) from the database. The `datasource_loop_variable` key holds the name of the key on each object in the datasource.
```

Comment:
Might be good to give a sentence or two about what data source items are and how they're gotten - from a database or smtn.